### PR TITLE
Allow rotation of profile picture #2610

### LINKED ISSUE #2610: Allow rotation of profile picture
when editing/uploading a profile picture, it will be useful to have this feature.

### PR DESCRIPTION
**Pull request by ashrayjain**
Fixes #2610 
(Duplicate of #4381 as it was accidentally closed)

### CODE REVIEW

File: src/test/resources/pages/studentProfilePictureModalDefault.html [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+                        <div class="center-block align-center">
+                            <form id="profilePictureUploadForm" method="post">
+                                <input id="studentPhoto" class="inline" name="studentprofilephoto" type="file">
+                                <p class="help-block">
+                                    Max Size: 5 MB
+                                </p>
+                                <button type="button" id="profileUploadPictureSubmit" class="btn btn-primary center-block" onclick="finaliseUploadPictureForm()">
+                                    Upload Picture
+                                </button>
+                                <input name="user" value="charlie.tmms.example" type="hidden">
```
Comment (thyageshm) [inline src/test/resources/pages/studentProfilePictureModalDefault.html]: hey could you confirm if the "charlie.tmms.example" here should have been converted to "${test.student2}"?

File: src/test/resources/pages/studentProfilePictureModalEdit.html [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                                <input name="user" type="hidden" value="${test.student2}" />
+                            </form>
+                        </div>
+                    </div>
+                    <div class="col-xs-8 profile-pic-edit-col border-left-gray">
+                        <div class="profile-pic-edit">
+                            <div class="profile-pic-edit-container">
+                                <div class="guillotine-window" style="width: 100%; height: auto; padding-top: 100%;">
+                                    <div class="guillotine-canvas" style="width: 100%; height: 100%; top: 0%; left: 0%;">
+                                        <img id="editableProfilePicture" src="/page/studentProfilePic?blob-key=encoded_gs_key:{*}&user=${test.student2}" style="perspective: 1000px; backface-visibility: hidden;" />
```
Comment (thyageshm) [inline src/test/resources/pages/studentProfilePictureModalEdit.html]: I am not sure if the blob-key here will fail in the staging server as the format is different in the live server. The "encoded_gs_key:" part does not exist I think... not sure though

**Comment by thyageshm** [comment]
some comments for you to take a look. Try running the test on the staging server.

**Comment by ashrayjain** [comment]
@thyageshm Indeed, on staging, the "encoded_gs_key" part doesnt exist. How should I fix this?

**Comment by thyageshm** [comment]
either you have to update godmode (in which case please wait for #4402) or do your comparison at a lower level element (like the div with class="profile-pic-edit-toolbar")

File: src/test/resources/pages/studentProfilePictureModalDefault.html [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+                        <div class="center-block align-center">
+                            <form id="profilePictureUploadForm" method="post">
+                                <input id="studentPhoto" class="inline" name="studentprofilephoto" type="file">
+                                <p class="help-block">
+                                    Max Size: 5 MB
+                                </p>
+                                <button type="button" id="profileUploadPictureSubmit" class="btn btn-primary center-block" onclick="finaliseUploadPictureForm()">
+                                    Upload Picture
+                                </button>
+                                <input name="user" value="${test.student2}" type="hidden">
```
Comment (thyageshm) [inline src/test/resources/pages/studentProfilePictureModalDefault.html]: did you put in the placeholder manually or using godmode?

Comment (ashrayjain) [inline src/test/resources/pages/studentProfilePictureModalDefault.html]: Did it manually

**Comment by thyageshm** [comment]
good to merge :-)

**Pull request closed without merging**